### DISCUSSION TITLE
[Forge 1.20.1] Config option to always guarantee 1 bounty board per village

### DISCUSSION THREAD

**Discussion by vizthex123** [ejektaflex/Bountiful#276] [Mod Ideas]
It would override the "board gen frequency" option - or perhaps replacing that option with a "boards per village" option?

Being able to configure certain villages to have certain decrees would be nice too (maybe even disabling duplicate decrees), as I've found villages with 2 - 3 of the same decree. Bit annoying.

**Comment by ejektaflex** [comment]
This is not possible due to how villages generate. We can only insert a gazebo into a pool with a varying weight and hope for the best.

If I could easily do that, I'd have done it already, I promise!

**Comment by vizthex123** [comment]
Oof.

So, I suppose you can't add a "don't add more than 1 board if 1 was generated" option?

**Comment by ejektaflex** [comment]
Nope. This is the random nature of how their pools work - they don't take into account which jigsaws have already been used. As such, the config option just weights the relative chance of a board showing up compared to a house. I'm not sure what the default house weight is - I think it's 4?

As for attempting to do what you ask - I have seen a mod successfully do this before, but it relied on the mod Mixin Extras. I did not want to rely on another dependency for Bountiful to work, so I refrained from using it. However, it's my understanding that Mixin Extras is bundled into both NeoForge and Fabric by default now, so I may look into it further soon.

That said, hooking into this algorithm is pretty invasive and complicated and I'm not sure if it might affect performance, so we'll have to see!

**Reply by vizthex123** [reply]
If it's not possible to add that's fine, I just wanted to suggest it. Not a programmer so idk the limitations of Forge & whatnot lol.
